src/main/kotlin/com/jetbrains/micropython/devices/PyboardDeviceProvider.kt

```diff
@@ -52,19 +56,33 @@ class PyboardDeviceProvider : MicroPythonDeviceProvider {
 
   override fun getRunCommandLineState(configuration: MicroPythonRunConfiguration,
                                       environment: ExecutionEnvironment): CommandLineState? {
+    val module = configuration.module ?: return null
     val facet = configuration.module?.microPythonFacet ?: return null
     val pythonPath = facet.pythonPath ?: return null
     val devicePath = facet.devicePath ?: return null
     val rootPath = configuration.project.basePath ?: return null
+    val file = StandardFileSystems.local().findFileByPath(configuration.path) ?: return null
+    val excludeRoots = ModuleRootManager.getInstance(module).excludeRoots
+    val excludes = excludeRoots
+            .asSequence()
+            .filter { VfsUtil.isAncestor(file, it, false) }
+            .filterNotNull()
+            .map { listOf("-X", it.path) }
+            .flatten()
+            .toList()
+    val excludeFiles = ModuleRootManager.getInstance(module).contentEntries
```

Comment:
This list of exclude patterns for module content entries duplicates the list of module excluded roots used above, so it's redundant. `excludeRoots` basically iterates over content entries and looks for `excludedRootFiles`.